### PR TITLE
Support testing versions separatelly

### PR DESCRIPTION
**Pull request by phracek**
This pull request tests nginx-versions separately

### CODE REVIEW

**Comment by phracek** [comment]
The pull request is blocked by this Testing Farm pull request https://github.com/sclorg/sclorg-testing-farm/pull/37

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
As an example you can see output here: https://github.com/phracek/nginx-container/pull/1 and https://github.com/phracek/nginx-container/actions/runs/2998828297

**Comment by phracek** [comment]
[test-all]

File: .github/workflows/container-tests.yml [hunk 1]
```diff
@@ -77,16 +77,39 @@ jobs:
...
+        shell: bash
+        id: vars
+        run: |
+          dockerfile="Dockerfile.${{ matrix.os_test }}"
+          if [[ ${{ matrix.os_test }} == "centos7" ]]; then
+            dockerfile="Dockerfile"
+          fi
+          echo "::set-output name=DOCKERFILE_NAME::${dockerfile}"
+
+      - name: Check ${{ matrix.version }}/${{ steps.vars.outputs.DOCKERFILE_NAME }} presents
```
Comment (zmiklank) [inline .github/workflows/container-tests.yml:90]:
```suggestion
      - name: Check ${{ matrix.version }}/${{ steps.vars.outputs.DOCKERFILE_NAME }} is present
```

File: .github/workflows/container-tests.yml [hunk 2]
```diff
@@ -77,16 +77,39 @@ jobs:
...
         with:
           api_key: ${{ secrets[matrix.api_key] }}
           git_url: ${{ matrix.tmt_repo }}
           git_ref: ${{ matrix.branch }}
           tf_scope: ${{ matrix.tf_scope }}
           tmt_plan_regex: ${{ matrix.tmt_plan }}
-          pull_request_status_name: ${{ matrix.context }}
-          variables: "REPO_URL=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY;REPO_NAME=$GITHUB_REPOSITORY;PR_NUMBER=${{ github.event.issue.number }};OS=${{ matrix.os_test }};TEST_NAME=test"
+          pull_request_status_name: "${{ matrix.context }} - ${{ matrix.version }}"
+          variables: "REPO_URL=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY;REPO_NAME=$GITHUB_REPOSITORY;PR_NUMBER=${{ github.event.issue.number }};VERSION=${{ matrix.version }};OS=${{ matrix.os_test }};TEST_NAME=test"
```
Comment (zmiklank) [inline .github/workflows/container-tests.yml:114]:
```suggestion
          variables: "REPO_URL=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY;REPO_NAME=$GITHUB_REPOSITORY;PR_NUMBER=${{ github.event.issue.number }};SINGLE_VERSION=${{ matrix.version }};OS=${{ matrix.os_test }};TEST_NAME=test"
```

File: .github/workflows/openshift-tests.yml [hunk 3]
```diff
@@ -68,16 +68,39 @@ jobs:
...
         with:
           api_key: ${{ secrets[matrix.api_key] }}
           git_url: ${{ matrix.tmt_repo }}
           git_ref: ${{ matrix.branch }}
           tf_scope: ${{ matrix.tf_scope }}
           tmt_plan_regex: ${{ matrix.tmt_plan }}
-          pull_request_status_name: ${{ matrix.context }}
-          variables: "REPO_URL=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY;REPO_NAME=$GITHUB_REPOSITORY;PR_NUMBER=${{ github.event.issue.number }};OS=${{ matrix.os_test }};TEST_NAME=${{ matrix.test_name }}"
+          pull_request_status_name: "${{ matrix.context }} - ${{ matrix.version }}"
+          variables: "REPO_URL=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY;REPO_NAME=$GITHUB_REPOSITORY;PR_NUMBER=${{ github.event.issue.number }};VERSION=${{ matrix.version }};OS=${{ matrix.os_test }};TEST_NAME=${{ matrix.test_name }}"
```
Comment (zmiklank) [inline .github/workflows/openshift-tests.yml:105]:
```suggestion
          variables: "REPO_URL=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY;REPO_NAME=$GITHUB_REPOSITORY;PR_NUMBER=${{ github.event.issue.number }};SINGLE_VERSION=${{ matrix.version }};OS=${{ matrix.os_test }};TEST_NAME=${{ matrix.test_name }}"
```

File: .github/workflows/openshift-tests.yml [hunk 4]
```diff
@@ -68,16 +68,39 @@ jobs:
...
+        shell: bash
+        id: vars
+        run: |
+          dockerfile="Dockerfile.${{ matrix.os_test }}"
+          if [[ ${{ matrix.os_test }} == "centos7" ]]; then
+            dockerfile="Dockerfile"
+          fi
+          echo "::set-output name=DOCKERFILE_NAME::${dockerfile}"
+
+      - name: Check ${{ matrix.version }}/${{ steps.vars.outputs.DOCKERFILE_NAME }} presents
```
Comment (zmiklank) [inline .github/workflows/openshift-tests.yml:81]:
```suggestion
      - name: Check if ${{ matrix.version }}/${{ steps.vars.outputs.DOCKERFILE_NAME }} is present
```

File: .github/workflows/container-tests.yml [hunk 5]
```diff
@@ -77,16 +77,39 @@ jobs:
...
+
+      - name: Check .exclude-${{ matrix.os_test }} file presents
+        id: check_exclude_file
+        uses: andstor/file-existence-action@v1
+        with:
+          files: "${{ matrix.version }}/.exclude-${{ matrix.os_test }}"
+
       # https://github.com/sclorg/testing-farm-as-github-action
-      - name: Schedule tests on external Testing Farm by Testing-Farm-as-github-action
+      - name: Schedule tests for ${{ matrix.context }} - ${{ matrix.version }}
```
Comment (zmiklank) [inline .github/workflows/container-tests.yml:103]: What about switching version and context here (and in similar places)? So it is the same as in the name of the job?

**Review by zmiklank — CHANGES_REQUESTED** [review]

File: .github/workflows/container-tests.yml (hunk 2 above)
Comment (phracek) [reply .github/workflows/container-tests.yml:114]: In TF we use VERSION https://github.com/sclorg/sclorg-testing-farm/blob/main/plans/c8s.fmf#L10
In my opinion, SINGLE_VERSION does not make sense here. But it is my opinion.

File: .github/workflows/container-tests.yml (hunk 5 above)
Comment (phracek) [reply .github/workflows/container-tests.yml:103]: It makes sense here.

File: .github/workflows/container-tests.yml (hunk 2 above)
Comment (zmiklank) [reply .github/workflows/container-tests.yml:114]: Right, sorry.

Comment (zmiklank) [reply .github/workflows/container-tests.yml:114]: Too many VERSIONS :).

**Comment by phracek** [comment]
@zmiklank Please review it one more time. Thank you.

**Review by zmiklank — APPROVED** [review]

**Comment by zmiklank** [comment]
But please squash the commits before merging(to two resulting commits?).

**Comment by phracek** [comment]
[test-all]

**Pull request merged**